warehouse/packaging/models.py

```diff
@@ -162,6 +162,11 @@ def __contains__(self, project):
             return True
 
 
+class LifecycleStatus(enum.StrEnum):
+    QuarantineEnter = "quarantine-enter"
+    QuarantineExit = "quarantine-exit"
```

Comment:
General question: this and the `lifecycle_status` field imply a state transition graph like the following:

```
None -> (QuarantineEnter <-> QuarantineExit)
```

...which makes sense to me, but might make adding other `LifecycleStatus` states a bit verbose (since each may need to have separate `Enter` and `Exit` states, and define whether a transition from one "color" exit to another "color" enter is valid).

Instead, would it make sense to have `LifecycleStatus.Nominal` (or similar) as a general "exit" state (and default state as well)? That would allow for recording of transitions without requiring each lifecycle state to have its own "exit" marker.

Reply:
See also https://github.com/pypi/warehouse/pull/16179#discussion_r1667152383 😉

Reply:
(I upgraded your state diagram 😉 )

Similar to the other comments di linked to, I think that these states should emerge as we discover them, and decide if the FSM complexity is worth it.

Reply:
> (I upgraded your state diagram 😉 )

Nice, I always forget GH MD can do Mermaid.

I totally missed @di's original comment, sorry about that 😅 -- I'll fold my thoughts in there!